docs/reference-guide/modules/events/pages/event-handlers.adoc

```diff
@@ -114,5 +420,365 @@ public class MyEventHandlingComponent {
+
+[source,java]
+----
+// GOOD: Focused handlers
+@EventHandler
+public void updateReadModel(OrderPlacedEvent event) {
+    // Only update read model
+}
+
+@EventHandler
+public void sendNotification(OrderPlacedEvent event, CommandDispatcher dispatcher) {
+    // Only send notification
+}
+
+// AVOID: Handler doing too many things
+@EventHandler
+public void handleEverything(OrderPlacedEvent event) {
+    // Update read model
+    // Send notifications
+    // Dispatch commands
+    // Log events
+    // etc.
+}
+----
+
+=== When to inject `ProcessingContext`
+
+Inject `ProcessingContext` as a parameter **only when you need it**.
+
+**You DON'T need ProcessingContext** when only using `CommandDispatcher` or `EventAppender`, as they're automatically context-aware when injected as parameters:
+
+[source,java]
+----
+@EventHandler
+public void on(OrderPlacedEvent event, CommandDispatcher dispatcher) {
+    // CommandDispatcher is already context-aware - correlation data flows automatically
+    dispatcher.send(new ReserveInventoryCommand(event.getOrderId()));
+}
+----
+
+**You DO need ProcessingContext** when:
+- Using components that require it explicitly (like `CommandGateway`)
+- Registering lifecycle callbacks (`onCommit`, `onCleanup`, etc.)
+- Accessing the message or its metadata
+- Passing context to your own services
+
+[source,java]
+----
+@EventHandler
+public void on(OrderPlacedEvent event,
+               ProcessingContext context,
+               CommandDispatcher dispatcher) {
+    // Use dispatcher (no context needed)
+    dispatcher.send(new ReserveInventoryCommand(event.getOrderId()));
+
+    // But also need context for lifecycle callback
+    context.onCommit(ctx -> auditLog.record(event));
+
+    // Or to pass to your own services
+    myService.processOrder(event, context);
+}
+----
+
+=== Inject dispatchers as parameters, not fields
+
+Always inject `EventAppender` and `CommandDispatcher` as method parameters:
+
+[source,java]
+----
+// CORRECT
+@EventHandler
+public void on(OrderPlacedEvent event, EventAppender appender) {
+    appender.append(new OrderConfirmedEvent(...));
+}
+
+// WRONG - will not be ProcessingContext-aware
+@Component
+public class OrderHandler {
+    private final EventAppender appender;  // Don't do this!
+
+    public OrderHandler(EventAppender appender) {
+        this.appender = appender;
+    }
+}
+----
+
+=== Handle events idempotently
+
+Event handlers may be invoked multiple times for the same event (due to retries, replays, or system failures).
+Design handlers to be idempotent:
+
+[source,java]
+----
+@EventHandler
+public void on(OrderPlacedEvent event) {
+    // Check if already processed
+    if (orderRepository.exists(event.getOrderId())) {
+        return;  // Already processed, skip
+    }
+
+    // Process event
+    orderRepository.save(new OrderView(event));
+}
+----
+
+=== Use result objects instead of exceptions for expected failures
```

Comment:
This section makes zero sense for event handlers, as those do not allow for return objects. Throwing an exception is **THE** reason to tell the Event Processor you're not happy with the outcome. Or, you'd have to schedule something to be picked up later on. Regardless, let's drop the "Use result objects instead of exceptions for expected failures" section entirely.